core-to-plc/src/Language/Plutus/CoreToPLC.hs

```diff
@@ -506,83 +511,94 @@ getDataCons tc' = sortConstructors tc' <$> extractDcs tc'
               GHC.NewTyCon{GHC.data_con=dc}    -> pure [dc]
           | otherwise = throwSd UnsupportedError $ "Type constructor:" GHC.<+> GHC.ppr tc
 
--- This is the creation of the Scott-encoded datatype type.
 -- See Note [Scott encoding of datatypes]
-convDataCons :: forall m. Converting m => GHC.TyCon -> [GHC.DataCon] -> m PLCType
-convDataCons tc dcs = do
-    abstracted <- flip (foldr (\tv acc -> mkTyLam tv acc)) (GHC.tyConTyVars tc) $ mkScottTy tc dcs
-
+-- | Make the type corresponding to a given 'TyCon'.
+mkTyCon :: forall m. Converting m => GHC.TyCon -> m PLCType
```

Comment:
This is not too important, but I've just realized I like the `conv` vs `mk` distinction with the former crossing language boundaries and the latter receiving Plutus Core entities and returning Plutus Core entities.

Reply:
I'd like that too, except it's not that consistent - the `mkLamAbsScoped` function for example takes a GHC var. I guess I've just made it inconsistent in a different way now :laughing: 

I'll have a think and see if I can think of a better naming strategy that's clear.